src/YourEasyRent/Services/TelegramCallbackHandler.cs

```diff
@@ -19,34 +22,47 @@ public class TelegramCallbackHandler : ITelegramCallbackHandler
         private Dictionary<string, IButtonHandler> _buttonHandlers;
         private Dictionary<long, BotState> _userResponsesToChat = new();
         private readonly ILogger<TelegramCallbackHandler> _logger;
-        // сделать еше один словать которыцй будет содержать  userId (long)  и связанный с ним словать  ключ - бренд и категория, значения - Мейбелин и тушь
-
-        public TelegramCallbackHandler(ITelegramBotClient botClient, ITelegramActionsHandler actionsHandler,ILogger<TelegramCallbackHandler> logger)
+        private readonly IProductRepository _productRepository;
+        private MenuStatus _currentMenuStatus = MenuStatus.Started;
+        private readonly IUserStateManager _userStateManager = new UserStateManager();
+
+        public TelegramCallbackHandler
+            (
+            ITelegramBotClient botClient,
+            ITelegramActionsHandler actionsHandler,
+            IProductRepository productRepository,
+            IUserStateManager userStateManager,
+            ILogger<TelegramCallbackHandler> logger
+            )
         {
             _botClient = botClient;
             _actionsHandler = actionsHandler;
+            _productRepository = productRepository;
             _buttonHandlers = new Dictionary<string, IButtonHandler>()
             {
-                { "MainMenu", new MainMenuButtonHandler(_botClient) },
-                { "BrandMenu", new BrandButtonHandler(_botClient) },
-                { "CategoryMenu", new CategoryButtonHandler(_botClient)  }
+                {"MainMenu" , new MainMenuButtonHandler(_botClient) },
```

Comment:
You can now bind to enum steps
{Steps.MainMenuStep, new MainMenuButtonHandler(_botClient) }